tests/functional/object/mcg/ui/test_mcg_ui.py

```diff
@@ -441,3 +442,181 @@ def test_bucket_create(self, setup_ui_class_factory):
+
+        ui_buckets = bucket_ui.get_buckets_list()
+        ui_bucket_count = len(ui_buckets)
+        logger.info(f"UI bucket count (current page): {ui_bucket_count}")
+
+        # Determine if we need to create more buckets
+        buckets_needed = max(0, 101 - initial_bucket_count)
+        if buckets_needed > 0:
+            logger.info(
+                f"Creating {buckets_needed} additional buckets to test pagination"
+            )
+
+            s3_buckets_to_create = buckets_needed // 2
+            obc_buckets_to_create = buckets_needed - s3_buckets_to_create
+
+            bucket_ui.create_multiple_buckets_ui(
+                s3_buckets=s3_buckets_to_create, obc_buckets=obc_buckets_to_create
+            )
+
+            bucket_ui.nav_object_storage_page()
+            bucket_ui.nav_buckets_page()
+
+            cli_buckets = list(mcg_obj.cli_list_all_buckets())
+            new_bucket_count = len(cli_buckets)
+            logger.info(f"New bucket count (from CLI): {new_bucket_count}")
+
+            assert (
+                new_bucket_count >= 101
+            ), f"Failed to create enough buckets for pagination. Current count: {new_bucket_count}"
+
+        assert (
+            bucket_ui.has_pagination_controls()
+        ), "Pagination controls not found despite having >100 buckets"
+        logger.info("Pagination controls are present")
+
+        # Get first page bucket names for comparison later
+        first_page_buckets = bucket_ui.get_buckets_list()
+        assert (
+            len(first_page_buckets) == 100
+        ), f"Expected 100 buckets on first page but found {len(first_page_buckets)}"
+        logger.info(f"First page has {len(first_page_buckets)} buckets")
+
+        assert bucket_ui.navigate_to_next_page(), "Failed to navigate to next page"
+        logger.info("Successfully navigated to the second page")
+
+        # Get second page bucket names to verify they are different
+        second_page_buckets = bucket_ui.get_buckets_list()
+        logger.info(f"Second page has {len(second_page_buckets)} buckets")
+
+        assert (
+            len(second_page_buckets) > 0
+        ), "Second page should have at least one bucket"
+
+        first_page_bucket_ids = set(first_page_buckets)
+        second_page_bucket_ids = set(second_page_buckets)
+        assert not first_page_bucket_ids.intersection(
+            second_page_bucket_ids
+        ), "Found duplicate buckets between pages"
+
+        assert (
+            bucket_ui.navigate_to_previous_page()
+        ), "Failed to navigate back to previous page"
+        logger.info("Successfully navigated back to the first page")
+
+        returned_first_page_buckets = bucket_ui.get_buckets_list()
+        returned_page_bucket_ids = set(returned_first_page_buckets)
+        assert (
+            returned_page_bucket_ids == first_page_bucket_ids
+        ), "Returned to first page but buckets don't match"
```

Comment:
I think it will be better to store test-to-result dict, so we can then do assertion on all values are True.
If you agree it will be add key-value pair to results dict instead of each assertion. 

This way we will collect result of every check and populate it, even when first "assertion" failed.
 
We did it other Complex tests, such as dashboard_validation: https://github.com/red-hat-storage/ocs-ci/blob/e17d1e9b3e944eebf9a75aa099901d93b8e7c8a3/ocs_ci/ocs/ui/validation_ui.py#L276